src/wallet/wallet.cpp

```diff
@@ -1914,13 +1935,18 @@ CAmount CWallet::GetBalance() const
             if (pcoin->IsTrusted())
                 nTotal += pcoin->GetAvailableCredit();
         }
+        nBalanceCache = nTotal;
+        fBalancesDirty = false;
```

Comment:
In commit "Add atomic cache for balances"

Maybe rename `fBalancesDirty` to `fBalanceDirty` since this only applies to the balance returned from CWallet::GetBalance and not the other balances.